modules/engines/engine-xssecurity/src/main/java/com/sap/xsk/xsaccess/ds/synchronizer/XSKSecuritySynchronizer.java

```diff
@@ -45,311 +44,301 @@
+                if (!existingXscPrivilegeDefinition.getDescription().equals(xskPrivilegeDefinition.getDescription())) {
+                    xskPrivilegeCoreService.updateXSKPrivileges(xskPrivilegeDefinition.getName(), xskPrivilegeDefinition.getDescription());
+                    logger.info("Synchronized a modified XSK Privilege [{}]", xskPrivilegeDefinition.getName());
+                }
+
+            }
+
+            PRIVILEGES_SYNCHRONIZED.add(xskPrivilegeDefinition.getName());
+        } catch (XSKPrivilegeException e) {
+            throw new SynchronizationException(e);
+        }
     }
-  }
-
-  /*
-   * (non-Javadoc)
-   * @see org.eclipse.dirigible.core.scheduler.api.AbstractSynchronizer#synchronizeRegistry()
-   */
-  @Override
-  protected void synchronizeRegistry() throws SynchronizationException {
-    logger.trace("Synchronizing Extension Points and Extensions from Registry...");
-
-    super.synchronizeRegistry();
-
-    logger.trace("Done synchronizing Extension Points and Extensions from Registry.");
-  }
-
-  /*
-   * (non-Javadoc)
-   * @see org.eclipse.dirigible.core.scheduler.api.AbstractSynchronizer#synchronizeResource(com.sap.xsk.hdb.ds.parser.XSKDataStructureParser
-   * repository.api.IResource)
-   */
-  @Override
-  protected void synchronizeResource(IResource resource) throws SynchronizationException {
-    String resourceName = resource.getName();
-    if (resourceName.equals(IXSKPrivilegeCoreService.XSK_FILE_EXTENSION_PRIVILEGE)) {
-      XSKPrivilegeArtifact xskPrivilegeArtifact = XSKPrivilegeArtifact.parse(resource.getContent());
-
-      // set name to use "path.to::Privilege" syntax
-      String[] splitPath = resource.getPath().split("/");
-      String path = String.join(".", Arrays.copyOfRange(splitPath, 3, splitPath.length - 1)); // remove /registry/public
-
-      for (XSKPrivilegeDefinition privilege : xskPrivilegeArtifact.divide()) {
-        privilege.setName(path + "::" + privilege.getName());
-        PRIVILEGES_TO_BE_PROCESSED.add(privilege);
-      }
+
+    /**
+     * Synchronize access.
+     *
+     * @param xskAccessDefinition the access definition
+     * @throws SynchronizationException the synchronization exception
+     */
+    private void synchronizeAccess(XSKAccessDefinition xskAccessDefinition) throws SynchronizationException {
+        try {
+            XSKAccessDefinition existingXSKAccessDefinition = xskAccessCoreService.getXSKAccessDefinition(xskAccessDefinition.getPath());
+            if (existingXSKAccessDefinition == null) {
+                xskAccessCoreService.createXSKAccessDefinition(xskAccessDefinition.getPath(), xskAccessDefinition.getAuthenticationMethodsAsList(),
+                    xskAccessDefinition.getHash(),
+                    xskAccessDefinition.isExposed(), xskAccessDefinition.getAuthorizationRolesAsList());
+                logger.info("Synchronized a new XSK Access definition [[{}]-[{}]] from location: {}",
```

Comment:
Couldn't these logger agruments be moved to variables or even move the whole `logger.info` in a separate method as it's not easily readable